### PR TITLE
Bug 1828479: Move topology search query from localstorage to url param

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes - https://issues.redhat.com/browse/ODC-3058

This PR -
- Updates the logic of saving topology filters in local storage.
- Now the topology saves only display filters in localstorage and saves search filter in the URL.
- Search filter is reset when changing the namespace.

Screencast - 

filters

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/redux/reducer.ts [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 export type State = Map<string, any>;
 
 export const getDefaultTopologyFilters = () => {
-  const filters = localStorage.getItem(TOPOLOGGY_FILTERS_LOCAL_STORAGE_KEY);
-  if (filters) {
-    return merge({}, DEFAULT_TOPOLOGY_FILTERS, JSON.parse(filters));
+  const displayFilters = localStorage.getItem(TOPOLOGY_DISPLAY_FILTERS_LOCAL_STORAGE_KEY);
+  const searchQuery = new URLSearchParams(window.location.search).get(TOPOLOGY_SEARCH_FILTER_KEY);
+
+  if (displayFilters && searchQuery) {
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/redux/reducer.ts:16]: What if we have one or the other but not both? We still should get these values.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/topology/redux/reducer.ts]: Okay, just fixed it. Earlier I had the implementation with `displayFilters || searchQuery` which resulted in error if `displayFilter` was not available and there is `searchQuery`. So fixed that one but didn't realize that the `displayFilter` will be reset with default in `LocalStorage` every time when `searchQuery` is not there.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/topology/redux/reducer.ts]: I've tested it for both scenarios now.

**Pull request merged**